tpgtools/overrides/containeraws/samples/cluster/basic_update.tf.tmpl

```diff
@@ -0,0 +1,84 @@
+data "google_container_aws_versions" "versions" {
+  project = "{{project}}"
+  location = "us-west1"
+}
+
+resource "google_container_aws_cluster" "primary" {
+  authorization {
+    admin_users {
+      username = "{{test_service_account}}"
+    }
+  }
+
+  aws_region = "{{aws_region}}"
+
+  control_plane {
+    aws_services_authentication {
+      role_arn          = "arn:aws:iam::{{aws_account_id}}:role/{{byo_multicloud_prefix}}-1p-dev-oneplatform"
+      role_session_name = "{{byo_multicloud_prefix}}-1p-dev-session"
+    }
+
+    config_encryption {
+      kms_key_arn = "arn:aws:kms:{{aws_region}}:{{aws_account_id}}:key/{{aws_database_encryption_key}}"
+    }
+
+    database_encryption {
+      kms_key_arn = "arn:aws:kms:{{aws_region}}:{{aws_account_id}}:key/{{aws_database_encryption_key}}"
+    }
+
+    iam_instance_profile = "{{byo_multicloud_prefix}}-1p-dev-controlplane"
+    subnet_ids           = ["{{aws_subnet}}"]
+    version   = "${data.google_container_aws_versions.versions.valid_versions[0]}"
+    instance_type        = "t3.medium"
+
+    main_volume {
+      iops        = 3000
+      kms_key_arn = "arn:aws:kms:{{aws_region}}:{{aws_account_id}}:key/{{aws_volume_encryption_key}}"
+      size_gib    = 10
+      volume_type = "GP3"
+    }
+
+    proxy_config {
+      secret_arn     = "arn:aws:secretsmanager:us-west-2:126285863215:secret:proxy_config20210824150329476300000001-ABCDEF"
+      secret_version = "12345678-ABCD-EFGH-IJKL-987654321098"
+    }
+
+    root_volume {
+      iops        = 3000
+      kms_key_arn = "arn:aws:kms:{{aws_region}}:{{aws_account_id}}:key/{{aws_volume_encryption_key}}"
+      size_gib    = 10
+      volume_type = "GP3"
+    }
+
+    security_group_ids = ["{{aws_security_group}}"]
+
+    ssh_config {
+      ec2_key_pair = "{{byo_multicloud_prefix}}-1p-dev-ssh"
+    }
+
+    tags = {
+      owner = "{{test_service_account}}"
+    }
+  }
+
+  fleet {
+    project = "{{project_number}}"
```

Comment:
Do these values accept project ID as well? I believe they *should* based on AIP 2510, so it may be worth raising w/ the API team if they do not

Reply:
>  Do these values accept project ID as well?
No, they currently don't accept project ID. For more context, AIP 2510 used to advocate always returning project number and the gkemulticloud API followed the old guidance. Accepting project ID could cause terraform to incorrectly detect a diff.
For our launch, we wanted to avoid making breaking API changes so we limited these values to only accept project numbers. This is something we're aware of and would consider supporting project ID post GA.

Reply:
Yeah, we talked about this extensively and they're aware it's not ideal for cleanliness of user configs, but it's the best we could do.